src/Maestro/Maestro.DataProviders/ConfigurationDataIngestor.cs

```diff
@@ -0,0 +1,588 @@
+                ingestedSubscriptions.Count, ingestedChannels.Count, ingestedDefaultChannels.Count);
+
+            var configurationSource = await _context.ConfigurationSources
+                .Where(cs => cs.Uri == repoUri && cs.Branch == branch)
+                .FirstOrDefaultAsync();
+
+            await IngestConfigurationInternal(
+                repoUri,
+                branch,
+                configurationSource,
+                ingestedChannels,
+                ingestedDefaultChannels,
+                ingestedSubscriptions);
+
+            _logger.LogInformation("Successfully completed configuration ingestion for repository {RepoUri} on branch {Branch}", repoUri, branch);
+        }
+        catch (DbUpdateException dbEx)
+        {
+            _logger.LogError(dbEx.InnerException ?? dbEx, "Failed to ingest configuration for repository {RepoUri} on branch {Branch} because of failed SQL constraints", repoUri, branch);
+            throw;
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Failed to ingest configuration for repository {RepoUri} on branch {Branch}", repoUri, branch);
+            throw;
+        }
+    }
+
+    private async Task IngestConfigurationInternal(
+        string repoUri,
+        string branch,
+        ConfigurationSource? configurationSource,
+        IReadOnlyList<ChannelYamlData> ingestedChannels,
+        IReadOnlyList<DefaultChannelYamlData> ingestedDefaultChannels,
+        IReadOnlyCollection<SubscriptionUpdateYamlData> ingestedSubscriptions)
+    {
+        try
+        {
+            if (configurationSource == null)
+            {
+                configurationSource = new ConfigurationSource
+                {
+                    Uri = repoUri,
+                    Branch = branch
+                };
+                configurationSource = (await _context.ConfigurationSources.AddAsync(configurationSource)).Entity;
+            }
+
+            Dictionary<Guid, Subscription> existingSubscriptions = _context.Subscriptions
+                .Where(sub => sub.ConfigurationSourceId == configurationSource.Id)
+                .ToDictionary(sub => sub.Id);
+
+            Dictionary<string, Channel> existingChannels = _context.Channels
+                .Where(c => c.ConfigurationSourceId == configurationSource.Id)
+                .ToDictionary(c => c.Name);
+
+            List<DefaultChannel> existingDefaultChannels = _context.DefaultChannels
+                .Where(dc => dc.ConfigurationSourceId == configurationSource.Id)
+                .ToList();
+
+            _logger.LogInformation("Found {ExistingSubscriptionCount} existing subscriptions, {ExistingChannelCount} existing channels, {ExistingDefaultChannelCount} existing default channels", 
+                existingSubscriptions.Count, existingChannels.Count, existingDefaultChannels.Count);
```

Comment:
```suggestion
            existingSubscriptions.Count, existingChannels.Count, existingDefaultChannels.Count);
```
no idea if I got the indent right, I took 4 whitespaces from it